### ISSUE TITLE
`clr-dg-row-detail` Components should expand to take the full width of the parent

### ISSUE THREAD

**Issue by hippee-lee** [vmware/clarity#1675]
**Select one ...**  (check one with "x")
```
[x] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior

The `clr-dg-row-detail` component should expand to take the full width of its parent (row) container. 

### Actual behavior
The `clr-dg-row-detail` component is only as wide as its content. 

See the datagrid under Custom Component Row Details in [this](https://plnkr.co/edit/uVsLLzqcgnjTc9SXUyVG?p=preview) plunker for a reproduction scenario. 

### Environment details

* **Angular version:** 4.x.x

* **Clarity version:** `^0.10.12`

* **OS and version:** All

* **Browser:** [all]

# Workaround
For now the user can fix this by adding `flex: 1 1 auto;`to the child thats not expanding to fill its container. 

See this plunker for the workaround: 
https://plnkr.co/edit/uVsLLzqcgnjTc9SXUyVG?p=preview

**Comment by jacobbutton** [comment]
Thanks for posting this here @hippee-lee. I never know if I should be posting this stuff in the issues here or over on stackoverflow. I don't want to clog up the issues here if it's not necessary.

**Comment by adityarb88** [comment]
@jacobbutton: This definitely was a good question for StackOverflow. While answering it, we found a bug and thats why moved it to Github :-) Thanks for filing this issue/question! Please note though that there is a workaround for this issue like @hippee-lee mentioned above. Just noting that so that you are not blocked by this bug.

**Comment by jacobbutton** [comment]
The workaround has already been implemented and is working wonderfully. Thanks again.

**Comment by jacobbutton** [comment]
@hippee-lee - This doesn't seem to be working with the new datagrid.  Am I doing something wrong?

https://stackblitz.com/edit/clarity-light-theme-v1-0-a5p7hh?file=src%2Fapp%2Fapp.component.html

**Comment by hippee-lee** [comment]
Hey @jacobbutton - this is more of a [StackOverflow](https://stackoverflow.com/questions/tagged/vmware-clarity) implementation question. In the future would you mind posting this type of question over there first? If we discover there is an actual bug or gap in the components implementation we will create a new issue on github for better tracking since this issue is already closed. And, posting these types of questions on SO is more visible to the community.

For the stackblitz you posted I'm not 100% sure what the goal is. It seems like you want one column for the datagrid row and two columns in the expandable details that flex with the datagrid width. Thats what [this solution](https://stackblitz.com/edit/gh-issue-1675-clr-dg-row-details-example) accomplishes. I modified your stackblitz to do that. If it doesn't meet your needs please submit a question on [StackOverflow](https://stackoverflow.com/questions/tagged/vmware-clarity) and tag it with the `vmware-clarity` tag. I am subscribed to that feed and get the notifications pretty quickly. Thank you.
